src/iqm/qiskit_iqm/iqm_naive_move_pass.py

```diff
@@ -12,408 +12,232 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Naive transpilation for the IQM Star architecture."""
-
-from datetime import datetime
 from typing import Optional, Union
+import warnings
 
-from qiskit import QuantumCircuit, user_config
-from qiskit.circuit import QuantumRegister, Qubit
-from qiskit.dagcircuit import DAGCircuit, DAGOpNode
-from qiskit.providers.models import BackendProperties
-from qiskit.transpiler import CouplingMap, Layout, TranspileLayout
+from pydantic_core import ValidationError
+from qiskit import QuantumCircuit, transpile
+from qiskit.converters import circuit_to_dag, dag_to_circuit
+from qiskit.dagcircuit import DAGCircuit
 from qiskit.transpiler.basepasses import TransformationPass
-from qiskit.transpiler.exceptions import TranspilerError
-from qiskit.transpiler.passmanager import PassManager
-from qiskit.transpiler.passmanager_config import PassManagerConfig
-from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager
-from qiskit.transpiler.target import Target
+from qiskit.transpiler.layout import Layout
+
+from iqm.iqm_client import Circuit as IQMClientCircuit
+from iqm.iqm_client.transpile import ExistingMoveHandlingOptions, transpile_insert_moves
 
-from .fake_backends.iqm_fake_backend import IQMFakeBackend
-from .iqm_circuit import IQMCircuit
-from .iqm_provider import IQMBackend
-from .iqm_transpilation import IQMOptimizeSingleQubitGates
-from .move_gate import MoveGate
+from .iqm_backend import IQMBackendBase, IQMTarget
+from .iqm_move_layout import generate_initial_layout
+from .qiskit_to_iqm import deserialize_instructions, serialize_instructions
 
 
 class IQMNaiveResonatorMoving(TransformationPass):
     """WIP Naive transpilation pass for resonator moving
 
     A naive transpiler pass for use with the Qiskit PassManager.
-    Although it requires a CouplingMap, Target, or Backend, it does not take this into account when adding MoveGates.
+    Although it requires a CouplingMap, Target, or Backend, it does not take this into account when adding MOVE gates.
```

Comment:
Doesn't this now take into account the target when adding MOVE gates? I'm not sure if all the other limitations mentioned in the docstring are valid anymore either.

Reply:
This is terribly outdated.